logstash-core/lib/logstash/config/mixin.rb

```diff
@@ -142,6 +142,11 @@ def config_init(params)
       instance_variable_set("@#{key}", value)
     end
 
+    # now that we know the parameters are valid, we can obfuscate the original copy
+    # of the parameters before storing them as an instance variable
+    self.class.validate_check_parameter_values(original_params)
```

Comment:
I would say, we should add an issue to cleanup this deb, are you ok with
that?

On Thu, Mar 31, 2016 at 3:37 PM Pier-Hugues Pellerin <
notifications@github.com> wrote:

> In logstash-core/lib/logstash/config/mixin.rb
> <https://github.com/elastic/logstash/pull/4952#discussion_r58053907>:
>
> > @@ -142,6 +142,11 @@ def config_init(params)
> >        instance_variable_set("@#{key}", value)
> >      end
> >
> > +    # now that we know the parameters are valid, we can obfuscate the original copy
> > +    # of the parameters before storing them as an instance variable
> > +    self.class.validate_check_parameter_values(original_params)
>
> I am OK with this fix event if we are actually validating twice the
> submitted values.
> The actual problem is we should split the value assignment from the
> validation.
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly or view it on GitHub
> <https://github.com/elastic/logstash/pull/4952/files/ad4641b6fdec0a4073212578ed22f82da392fb6d#r58053907>
>